### ISSUE TITLE
Column in logger_logs for rowcounts (SQL%ROWCOUNT)

### ISSUE THREAD

**Issue by gbreazu** [OraOpenSource/Logger#152]
Hi,

it would be very helpful to have in table LOGGER_LOGS a column dedicated only to logging rowcounts (and the corresponding parameter in the LOG procedure), would make reporting much easier than having to parse the content of column "Extra". Maybe even a couple of columns (separated for insert, update and delete).

I'm using logger not only for "classical" instrumentation but also for logging of ETL-Processes (Data Warehouse batch processing).

Whant do you think about this enhancement request? Do you see a chance of implenting it?

Thank you,
Georg

**Comment by martindsouza** [comment]
@gbreazu Good idea. I've scheduled this as an enhangement for the the next release (date TBD). I will need to check any performance impacts before implementing. Pending any issues we could create a new column for it.

No parameter would be added to the `log` procedure as we can do this automatically behind the sense.

I don't see how we'd have separate columns for insert, update, and delete. All we'd be logging is `sql%rowcount`. Did you have a different idea on how to identify this?

**Comment by gbreazu** [comment]
Hi Martin,

thank you for the quick reply and tagging it as an enhancement for the next release :-).

Great if it can be done transparently behind the scene, haven't thought about this possibility due to the "volatility" of SQL%ROWCOUNT's content.

You are right, separate columns, according to the type of DML executed, don't make normally sense (and it wouldn't be normalized :-)). You have only one type of DML per logger_logs record and can "manually" record the type of DML performed in the EXTRA column. I proposed the separate columns for the special case of MERGE statements, where you can have up to three different DMLs per DML (e.g. logger_logs record) and (at least in a DWH environment) sometimes it is a requirement to log all three MERGE "components". But as it can't be done via SQL%ROWCOUNT you would (probably) need new parameters etc. It would have been nice to have, but probably too much of a special case to extend LOGGER this way and to cumbersome to use in most use cases. You can still use the new mechanism and have the SQL$ROWCOUNT of the merge recorded "automatically" in the new column and, if needed, store the value split up in the component DMLs into the EXTRA column.

Regards,
Georg

**Comment by jeffreykemp** [comment]
My preference actually would be to have this as an optional parameter to the logger call, otherwise there will be a lot of values in the SQL%ROWCOUNT column that are meaningless. e.g. the parameter could be p_sql_action which the developer would set to "inserted mytable", "updated mytable", etc. and if p_sql_action is not null, logger would record the SQL%ROWCOUNT.

**Comment by tomlieber** [comment]
i build it in the debug levelmode .. adding new column processedrows and change ins_logger_log and the log procedures in the logger package and package body.. this works fine for me. another workaround is to set the application_info or put the value in the parameter variables.. 
Good Package.. we build in the last years also loggers, but not so fine ..
